pkg/telemetry/cassandra_agent/cassandra_agent_config.go

```diff
@@ -0,0 +1,148 @@
+package cassandra_agent
+
+import (
+	"context"
+	"path/filepath"
+	"time"
+
+	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	telemetryapi "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/annotations"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/labels"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/result"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	agentConfigLocation = "/opt/management-api/metric-collector.yaml"
+	defaultAgentConfig  = telemetryapi.CassandraAgentSpec{
+		Endpoint: telemetryapi.Endpoint{
+			Port:    "9000",
+			Address: "127.0.0.1",
+		},
+	}
+)
+
+type Configurator struct {
+	TelemetrySpec telemetryapi.TelemetrySpec
+	Kluster       *k8ssandraapi.K8ssandraCluster
+	Ctx           context.Context
+	RemoteClient  client.Client
+	RequeueDelay  time.Duration
+}
+
+func (c Configurator) GetTelemetryAgentConfigMap() (*corev1.ConfigMap, error) {
+	var yamlData []byte
+	var err error
+	if c.TelemetrySpec.Cassandra != nil {
+		yamlData, err = yaml.Marshal(&c.TelemetrySpec.Cassandra)
+		if err != nil {
+			return &corev1.ConfigMap{}, err
+		}
+	} else {
+		yamlData, err = yaml.Marshal(&defaultAgentConfig)
+		if err != nil {
+			return &corev1.ConfigMap{}, err
+		}
+	}
+
+	cm := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: c.Kluster.Namespace,
+			Name:      c.Kluster.Name + "-metrics-agent-config",
+		},
+		Data: map[string]string{filepath.Base(agentConfigLocation): string(yamlData)},
+	}
+	return &cm, nil
+}
+
+func (c Configurator) ReconcileTelemetryAgentConfig(dc *cassdcapi.CassandraDatacenter) result.ReconcileResult {
+	//Reconcile the agent's ConfigMap
+	desiredCm, err := c.GetTelemetryAgentConfigMap()
+	if err != nil {
+		return result.Error(err)
+	}
+	cmObjectKey := types.NamespacedName{Name: c.Kluster.Name + "-metrics-agent-config",
+		Namespace: c.Kluster.Namespace}
```

Comment:
Same here, the configmap will be created for each datacenter, instead of one per cluster.

The configmap name should be extracted to a function so that we don't repeat it in multiple places.